src/System.Management.Automation/engine/hostifaces/LocalPipeline.cs

```diff
@@ -625,18 +625,16 @@ private void InvokeThreadProc()
                 SetPipelineState(PipelineState.Failed, ex);
                 SetHadErrors(true);
             }
-#if !CORECLR // No ThreadAbortException In CoreCLR
             catch (ThreadAbortException ex)
```

Comment:
`Thread.Abort()` is not supported in .NET Core.
According to the [doc](https://docs.microsoft.com/en-us/dotnet/api/system.threading.thread.abort?view=netcore-3.0), this method will throw `PlatformNotSupportedException` on .NET Core.

Given that, I suggest we just remove the `catch (ThreadAbortException ex)` block.

Reply:
I wonder that ThreadAbortException is not marked with `Obsolute()`.

I search ThreadAbortException in our code base and remove all related code.